CMakeLists.txt

```
@@ -47,6 +47,11 @@ include(CMakeDependentOption)
 include(CMakePackageConfigHelpers)
 include(FeatureSummary)
 
+# We need to enable C++ before trying to check for coverage
+if(WITH_GTEST OR WITH_BENCHMARKS)
```

```suggestion
if(WITH_GTEST OR WITH_FUZZERS OR WITH_BENCHMARKS)
```

test/fuzz can also use C++:

https://github.com/zlib-ng/zlib-ng/blob/e4a31e0538277a7960d5e3e78023db38fa301f9a/test/fuzz/CMakeLists.txt#L4

---

Flags enabled by default are defined later this ``if``
For example ``WITH_GTEST``.

I think we need reorder code like this:

https://github.com/phprus/zlib-ng/commit/b0832f2e49ba46fe525c30c14d447f8c558aa4c9

---

I'd like to keep the patch as short as possible, so moving only the "used" options before the check...